Loop/Managers/AlertPermissionsChecker.swift

```diff
@@ -64,92 +41,196 @@ class AlertPermissionsChecker {
+        case hasIssuedRiskMitigatingAlert = "com.loopkit.Loop.HasIssuedRiskMitigatingAlert"
+        case hasIssuedScheduledDeliveryEnabledAlert = "com.loopkit.Loop.HasIssuedScheduledDeliveryEnabledAlert"
     }
     
-    var hasIssuedNotificationsPermissionsAlert: Bool {
+    var hasIssuedRiskMitigatingAlert: Bool {
         get {
-            return object(forKey: Key.hasIssuedNotificationsPermissionsAlert.rawValue) as? Bool ?? false
+            return object(forKey: Key.hasIssuedRiskMitigatingAlert.rawValue) as? Bool ?? false
         }
         set {
-            set(newValue, forKey: Key.hasIssuedNotificationsPermissionsAlert.rawValue)
+            set(newValue, forKey: Key.hasIssuedRiskMitigatingAlert.rawValue)
         }
     }
-    
-    var hasIssuedCriticalAlertPermissionsAlert: Bool {
+
+    var hasIssuedScheduledDeliveryEnabledAlert: Bool {
+        get {
+            return object(forKey: Key.hasIssuedScheduledDeliveryEnabledAlert.rawValue) as? Bool ?? false
+        }
+        set {
+            set(newValue, forKey: Key.hasIssuedScheduledDeliveryEnabledAlert.rawValue)
+        }
+    }
+}
+
+struct NotificationCenterSettingsFlags: OptionSet {
+    let rawValue: Int
+
+    static let none = NotificationCenterSettingsFlags([])
+    static let notificationsDisabled = NotificationCenterSettingsFlags(rawValue: 1 << 0)
+    static let criticalAlertsDisabled = NotificationCenterSettingsFlags(rawValue: 1 << 1)
+    static let timeSensitiveNotificationsDisabled = NotificationCenterSettingsFlags(rawValue: 1 << 2)
+    static let scheduledDeliveryEnabled = NotificationCenterSettingsFlags(rawValue: 1 << 3)
+
+    static let riskMitigating: NotificationCenterSettingsFlags = [ .notificationsDisabled, .criticalAlertsDisabled, .timeSensitiveNotificationsDisabled ]
+}
+
+extension NotificationCenterSettingsFlags {
+    var notificationsDisabled: Bool {
+        get {
+            contains(.notificationsDisabled)
+        }
+        set {
+            update(.notificationsDisabled, newValue)
+        }
+    }
+    var criticalAlertsDisabled: Bool {
+        get {
+            contains(.criticalAlertsDisabled)
+        }
+        set {
+            update(.criticalAlertsDisabled, newValue)
+        }
+    }
+    var timeSensitiveNotificationsDisabled: Bool {
+        get {
+            contains(.timeSensitiveNotificationsDisabled)
+        }
+        set {
+            update(.timeSensitiveNotificationsDisabled, newValue)
+        }
+    }
+    var scheduledDeliveryEnabled: Bool {
         get {
-            return object(forKey: Key.hasIssuedCriticalAlertPermissionsAlert.rawValue) as? Bool ?? false
+            contains(.scheduledDeliveryEnabled)
         }
         set {
-            set(newValue, forKey: Key.hasIssuedCriticalAlertPermissionsAlert.rawValue)
+            update(.scheduledDeliveryEnabled, newValue)
+        }
+    }
+    var isRiskMitigating: Bool {
```

Comment:
[nit] Yea, this seems backwards to me.

Reply:
I couldn't think of a good term for the new "ganged-together" set of conditions other than "Risk Mitigating".  I'm open to other suggestions.  (Or, is your concern with the term "mitigating", because that implies that it is a verb that is mitigating the risk?)